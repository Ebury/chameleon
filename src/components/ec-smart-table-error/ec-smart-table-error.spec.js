import { mount } from '@vue/test-utils';
import EcSmartTableError from './ec-smart-table-error.vue';

function mountEcSmartTableError(props, mountOpts) {
  return mount(EcSmartTableError, {
    propsData: {
      title: 'Random Title',
      errorMessage: 'Random error message',
      ...props,
    },
    ...mountOpts,
  });
}

describe('EcSmartTableError', () => {
  it('should render properly', () => {
    const wrapper = mountEcSmartTableError();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render the title if title prop was not given', () => {
    const wrapper = mountEcSmartTableError({ title: null });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render default error message', () => {
    const wrapper = mountEcSmartTableError({ errorMessage: undefined });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render given error slot', () => {
    const wrapper = mountEcSmartTableError({ errorMessage: 'Test error message' }, {
      scopedSlots: {
        error: '<strong>Error Slot: {{props.errorMessage}}</strong>',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render given filter slot', () => {
    const wrapper = mountEcSmartTableError(null, {
      scopedSlots: {
        filter: '<div>Custom filter</div>',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render given header-actions slot', () => {
    const wrapper = mountEcSmartTableError(null, {
      scopedSlots: {
        'header-actions': '<div>Header Actions</div>',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render given header-actions slot with the props', () => {
    const wrapper = mountEcSmartTableError(null, {
      scopedSlots: {
        'header-actions': '<div slot-scope="{{ total, items, error, loading }}">Header Actions total: {{ total }}, items: {{ items }}, error: {{ error }}, loading: {{ loading }}</div>',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render given filter and header-actions slots', () => {
    const wrapper = mountEcSmartTableError(null, {
      scopedSlots: {
        filter: '<div>Custom filter</div>',
        'header-actions': '<div>Header Actions</div>',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
