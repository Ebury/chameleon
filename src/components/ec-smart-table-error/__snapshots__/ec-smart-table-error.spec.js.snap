// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcSmartTableError should not render the title if title prop was not given 1`] = `
<div
  class="ec-smart-table-error"
  data-test="ec-smart-table-error"
>
  <div>
    <!---->
     
    <!---->
  </div>
   Random error message
</div>
`;

exports[`EcSmartTableError should render default error message 1`] = `
<div
  class="ec-smart-table-error"
  data-test="ec-smart-table-error"
>
  <div>
    <div
      class="ec-smart-table-heading-title"
    >
      Random Title
    </div>
     
    <!---->
  </div>
   Unexpected error while fetching data
</div>
`;

exports[`EcSmartTableError should render given error slot 1`] = `
<div
  class="ec-smart-table-error"
  data-test="ec-smart-table-error"
>
  <div>
    <div
      class="ec-smart-table-heading-title"
    >
      Random Title
    </div>
     
    <!---->
  </div>
   
  <strong>
    Error Slot: Test error message
  </strong>
</div>
`;

exports[`EcSmartTableError should render given filter and header-actions slots 1`] = `
<div
  class="ec-smart-table-error"
  data-test="ec-smart-table-error"
>
  <div>
    <div
      class="ec-smart-table-heading-title"
    >
      Random Title
    </div>
     
    <div
      class="ec-smart-table-heading"
    >
      <div
        class="ec-smart-table-heading__filter"
      >
        <div>
          Custom filter
        </div>
      </div>
       
      <div
        class="ec-smart-table-heading__actions"
      >
        <div>
          Header Actions
        </div>
      </div>
    </div>
  </div>
   Random error message
</div>
`;

exports[`EcSmartTableError should render given filter slot 1`] = `
<div
  class="ec-smart-table-error"
  data-test="ec-smart-table-error"
>
  <div>
    <div
      class="ec-smart-table-heading-title"
    >
      Random Title
    </div>
     
    <div
      class="ec-smart-table-heading"
    >
      <div
        class="ec-smart-table-heading__filter"
      >
        <div>
          Custom filter
        </div>
      </div>
       
      <!---->
    </div>
  </div>
   Random error message
</div>
`;

exports[`EcSmartTableError should render given header-actions slot 1`] = `
<div
  class="ec-smart-table-error"
  data-test="ec-smart-table-error"
>
  <div>
    <div
      class="ec-smart-table-heading-title"
    >
      Random Title
    </div>
     
    <div
      class="ec-smart-table-heading"
    >
      <!---->
       
      <div
        class="ec-smart-table-heading__actions"
      >
        <div>
          Header Actions
        </div>
      </div>
    </div>
  </div>
   Random error message
</div>
`;

exports[`EcSmartTableError should render given header-actions slot with the props 1`] = `
<div
  class="ec-smart-table-error"
  data-test="ec-smart-table-error"
>
  <div>
    <div
      class="ec-smart-table-heading-title"
    >
      Random Title
    </div>
     
    <div
      class="ec-smart-table-heading"
    >
      <!---->
       
      <div
        class="ec-smart-table-heading__actions"
      >
        <div>
          Header Actions total: 0, items: [], error: Random error message, loading: false
        </div>
      </div>
    </div>
  </div>
   Random error message
</div>
`;

exports[`EcSmartTableError should render properly 1`] = `
<div
  class="ec-smart-table-error"
  data-test="ec-smart-table-error"
>
  <div>
    <div
      class="ec-smart-table-heading-title"
    >
      Random Title
    </div>
     
    <!---->
  </div>
   Random error message
</div>
`;
